--- nginx.spec.org	2018-01-24 17:54:59.000000000 +0100
+++ nginx.spec	2018-01-24 19:00:58.000000000 +0100
@@ -48,7 +48,7 @@
 # end of distribution specific definitions
 
 %define main_version 1.12.2
-%define main_release 1%{?dist}.ngx
+%define main_release 1%{?dist}.COLUNDRUM
 
 %define bdir %{_builddir}/%{name}-%{main_version}
 
@@ -59,8 +59,9 @@
 
 Summary: High performance web server
 Name: nginx
-Version: %{main_version}
+Version: 1.12.2
 Release: %{main_release}
+Packager: COLUNDRUM <colundrum@users.noreply.github.com>
 Vendor: Nginx, Inc.
 URL: http://nginx.org/
 Group: %{_group}
@@ -78,6 +79,14 @@
 Source11: nginx-debug.service
 Source12: COPYRIGHT
 Source13: nginx.check-reload.sh
+Source14: headers-more-nginx-module-0.33.tar.gz
+Source15: naxsi-0.55.3.tar.gz
+Source16: http-auth-pam-1.5.1.tar.gz
+Source17: http-cache-purge-2.3.tar.gz
+Source18: http-fancyindex-0.4.2.tar.gz
+Source19: nchan-1.1.14.tar.gz
+Source20: http-uploadprogress-0.9.2.tar.gz
+Source21: http-upstream-fair.zip
 
 License: 2-clause BSD-like license
 
@@ -104,14 +113,116 @@
     -e 's|%%PROVIDES%%|nginx-debug|g' < %{SOURCE2} > nginx-debug.init
 
 %build
-./configure %{BASE_CONFIGURE_ARGS} \
+./configure --prefix=%{_sysconfdir}/nginx \
+    --sbin-path=%{_sbindir}/nginx \
+    --modules-path=%{_libdir}/nginx/modules \
+    --conf-path=%{_sysconfdir}/nginx/nginx.conf \
+    --error-log-path=%{_localstatedir}/log/nginx/error.log \
+    --http-log-path=%{_localstatedir}/log/nginx/access.log \
+    --pid-path=%{_localstatedir}/run/nginx.pid \
+    --lock-path=%{_localstatedir}/run/nginx.lock \
+    --http-client-body-temp-path=%{_localstatedir}/cache/nginx/client_temp \
+    --http-proxy-temp-path=%{_localstatedir}/cache/nginx/proxy_temp \
+    --http-fastcgi-temp-path=%{_localstatedir}/cache/nginx/fastcgi_temp \
+    --http-uwsgi-temp-path=%{_localstatedir}/cache/nginx/uwsgi_temp \
+    --http-scgi-temp-path=%{_localstatedir}/cache/nginx/scgi_temp \
+    --user=%{nginx_user} \
+    --group=%{nginx_group} \
+    --with-compat \
+    --with-file-aio \
+    --with-threads \
+    --with-http_addition_module \
+    --with-http_auth_request_module \
+    --with-http_dav_module \
+    --with-http_flv_module \
+    --with-http_gunzip_module \
+    --with-http_gzip_static_module \
+    --with-http_mp4_module \
+    --with-http_random_index_module \
+    --with-http_realip_module \
+    --with-http_secure_link_module \
+    --with-http_slice_module \
+    --with-http_ssl_module \
+    --with-http_stub_status_module \
+    --with-http_sub_module \
+    --with-http_v2_module \
+    --with-mail \
+    --with-mail_ssl_module \
+    --with-stream \
+    --with-stream_realip_module \
+    --with-stream_ssl_module \
+    --with-stream_ssl_preread_module \
+    --with-pcre-jit \
+    --with-http_geoip_module=dynamic \
+    --with-http_image_filter_module=dynamic \
+    --with-http_perl_module=dynamic \
+    --with-http_xslt_module=dynamic \
+    --add-module=../../SOURCES/naxsi-0.55.3/naxsi_src \
+    --add-module=../../SOURCES/headers-more-nginx-module-0.33 \
+    --add-module=../../SOURCES/ngx_http_auth_pam_module-1.5.1 \
+    --add-module=../../SOURCES/http-cache-purge-2.3 \
+    --add-module=../../SOURCES/http-fancyindex-0.4.2 \
+    --add-module=../../SOURCES/nchan \
+    --add-module=../../SOURCES/http-uploadprogress \
+    --add-module=../../SOURCES/http-upstream-fair-master \
     --with-cc-opt="%{WITH_CC_OPT}" \
     --with-ld-opt="%{WITH_LD_OPT}" \
     --with-debug
 make %{?_smp_mflags}
 %{__mv} %{bdir}/objs/nginx \
     %{bdir}/objs/nginx-debug
-./configure %{BASE_CONFIGURE_ARGS} \
+./configure --prefix=%{_sysconfdir}/nginx \
+    --sbin-path=%{_sbindir}/nginx \
+    --modules-path=%{_libdir}/nginx/modules \
+    --conf-path=%{_sysconfdir}/nginx/nginx.conf \
+    --error-log-path=%{_localstatedir}/log/nginx/error.log \
+    --http-log-path=%{_localstatedir}/log/nginx/access.log \
+    --pid-path=%{_localstatedir}/run/nginx.pid \
+    --lock-path=%{_localstatedir}/run/nginx.lock \
+    --http-client-body-temp-path=%{_localstatedir}/cache/nginx/client_temp \
+    --http-proxy-temp-path=%{_localstatedir}/cache/nginx/proxy_temp \
+    --http-fastcgi-temp-path=%{_localstatedir}/cache/nginx/fastcgi_temp \
+    --http-uwsgi-temp-path=%{_localstatedir}/cache/nginx/uwsgi_temp \
+    --http-scgi-temp-path=%{_localstatedir}/cache/nginx/scgi_temp \
+    --user=%{nginx_user} \
+    --group=%{nginx_group} \
+    --with-compat \
+    --with-file-aio \
+    --with-threads \
+    --with-http_addition_module \
+    --with-http_auth_request_module \
+    --with-http_dav_module \
+    --with-http_flv_module \
+    --with-http_gunzip_module \
+    --with-http_gzip_static_module \
+    --with-http_mp4_module \
+    --with-http_random_index_module \
+    --with-http_realip_module \
+    --with-http_secure_link_module \
+    --with-http_slice_module \
+    --with-http_ssl_module \
+    --with-http_stub_status_module \
+    --with-http_sub_module \
+    --with-http_v2_module \
+    --with-mail \
+    --with-mail_ssl_module \
+    --with-stream \
+    --with-stream_realip_module \
+    --with-stream_ssl_module \
+    --with-stream_ssl_preread_module \
+    --with-pcre-jit \
+    --with-http_geoip_module=dynamic \
+    --with-http_image_filter_module=dynamic \
+    --with-http_perl_module=dynamic \
+    --with-http_xslt_module=dynamic \
+    --add-module=../../SOURCES/naxsi-0.55.3/naxsi_src \
+    --add-module=../../SOURCES/headers-more-nginx-module-0.33 \
+    --add-module=../../SOURCES/ngx_http_auth_pam_module-1.5.1 \
+    --add-module=../../SOURCES/http-cache-purge-2.3 \
+    --add-module=../../SOURCES/http-fancyindex-0.4.2 \
+    --add-module=../../SOURCES/nchan \
+    --add-module=../../SOURCES/http-uploadprogress \
+    --add-module=../../SOURCES/http-upstream-fair-master \
     --with-cc-opt="%{WITH_CC_OPT}" \
     --with-ld-opt="%{WITH_LD_OPT}"
 make %{?_smp_mflags}
@@ -312,6 +423,9 @@
 fi
 
 %changelog
+* Wed Jan 24 2018 COLUNDRUM <colundrum@users.noreply.github.com>
+- 1.12.2 custom build
+
 * Tue Oct 17 2017 Konstantin Pavlov <thresh@nginx.com>
 - 1.12.2
 
